react-app/src/api/distributionAPI.ts

```diff
@@ -306,6 +307,18 @@ export const useDistributionAPI = (): IDistributionAPI => {
     [query.mint, staking]
   );
 
+  const getBatchValidatorExpectedReturn = useCallback(
+    async (addresses: string[]) => {
+      const expectedReturnPromises: Promise<number>[] = [];
+      for (const address of addresses) {
+        expectedReturnPromises.push(getValidatorExpectedReturn(address));
```

Comment:
```
      const [annualProvisions, pool, validator] = await Promise.all([
        query.mint.annualProvisions(),
        staking.getPool(),
        staking.getValidator(address),
      ]);
```

possible for the pool and annualProvisions to change in the process if we can this in batch?  🤔

Reply:
Might ah, I've updated it ensure the same pool and annual provision values are used